packages/extension-dapp/src/bundle.ts

```diff
@@ -47,9 +47,9 @@ let web3EnablePromise: Promise<InjectedExtension[]> | null = null;
 
 export { isWeb3Injected, web3EnablePromise };
 
-function getWindowExtensions (originName: string): Promise<[InjectedExtensionInfo, Injected | void][]> {
+function getWindowExtension (originName: string, extensionName?: string): Promise<[InjectedExtensionInfo, Injected | void][]> {
   return Promise.all(
-    Object.entries(win.injectedWeb3).map(
+    Object.entries(win.injectedWeb3).filter(() => extensionName ? win.injectedWeb3[extensionName] : true).map(
```

Comment:
This is not quite correct. You don't actually filter the entries based on the name retrieved.

```js
Object.entries(win.injectedWeb3).filter(([name]) => !extensionName || name === extensionName)
```